tools/index_construction.cpp

```diff
@@ -45,8 +45,7 @@ void run(std::ifstream &input,
             << " seconds" << std::endl;
 
```

Comment:
Since your neat "run" template function seems to get called in a lot of contexts, might be worth breaking it out into a utility.